microcosm_logging/tests/test_decorators.py

```diff
@@ -15,7 +19,43 @@ class TestClass(object):
     pass
 
 
+@logger
+class TestContextClass(object):
+
+    def function(self):
+        self.logger.info("success!")
+
+
 def test_using_class_logger_works():
     instance = TestClass()
 
     assert_that(instance.logger, is_(equal_to(getLogger(instance.__class__.__name__))))
+
+
+def test_using_context_logger_works():
+    instance = TestContextClass()
+    context_func = Mock(return_value={'some': 'context'})
+    wrapped = context_logger(context_func, instance.function, instance)
```

Comment:
OK, I think I'm just missing the mechanism for injecting arguments. If the router is injecting the logger, where does its `extra` arguments come from?